api/v1alpha1/contentlibraryitemimportrequest_types.go

```diff
@@ -0,0 +1,164 @@
+}
+
+// ContentLibraryItemImportRequestSpec defines the desired state of a
+// ContentLibraryItemImportRequest.
+type ContentLibraryItemImportRequestSpec struct {
+	// Source is the source of the import request which includes an external URL
+	// pointing to a VM image template.
+	// +required
+	Source ContentLibraryItemImportRequestSource `json:"source"`
+
+	// Target is the target of the import request which includes the content library item
+	// information and a ContentLibrary resource.
+	// +required
+	Target ContentLibraryItemImportRequestTarget `json:"target"`
+
+	// TTLSecondsAfterFinished is the time-to-live duration for how long this
+	// resource will be allowed to exist once the import operation
+	// completes. After the TTL expires, the resource will be automatically
+	// deleted without the user having to take any direct action.
+	// If this field is unset then the request resource will not be
+	// automatically deleted. If this field is set to zero then the request
+	// resource is eligible for deletion immediately after it finishes.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	TTLSecondsAfterFinished *int64 `json:"ttlSecondsAfterFinished,omitempty"`
+}
+
+// ContentLibraryItemFileUploadStatus indicates the upload status of files belonging to the template.
+type ContentLibraryItemFileUploadStatus struct {
+	// InProgress lists the names of files that are being imported into vSphere.
+	// +optional
+	InProgress []string `json:"inProgress,omitempty"`
+
+	// Completed lists the names of files that have been successfully imported into vSphere.
+	// +optional
+	Completed []string `json:"completed,omitempty"`
+}
+
+// ContentLibraryItemImportRequestStatus defines the observed state of a
+// ContentLibraryItemImportRequest.
+type ContentLibraryItemImportRequestStatus struct {
+	// ItemRef is the reference to the target ContentLibraryItem resource of the import request.
+	// +optional
+	ItemRef *LocalObjectRef `json:"itemRef,omitempty"`
+
+	// CompletionTime represents time when the request was completed.
+	// The value of this field should be equal to the value of the
+	// LastTransitionTime for the status condition Type=Complete.
+	// +optional
+	CompletionTime metav1.Time `json:"completionTime,omitempty"`
+
+	// StartTime represents time when the request was acknowledged by the
+	// controller.
+	// +optional
+	StartTime metav1.Time `json:"startTime,omitempty"`
+
+	// Ready is set to true only when the content library item has been
+	// created and the template is imported successfully in vSphere
+	// and the new ContentLibraryItem resource is ready.
+	// Readiness is determined by waiting until there is status condition
+	// Type=Complete and ensuring it and all other status conditions present
+	// have a Status=True. The conditions present will be:
+	//   * SourceValid
+	//   * TargetValid
+	//   * ContentLibraryItemCreated
+	//   * TemplateUploaded
+	//   * ContentLibraryItemReady
+	//   * Complete
```

Comment:
Then this allows one import request instance to produce multiple clitem/vmi instances, a simpler option is to make the source and target to be immutable.

Reply:
As discussed offline, we can block the CR from being updated if the SourceValid and TargetValid conditions are true. This would ensure that a single import request would create a single CLItem.